src/controllers/auth/signup.ts

```diff
@@ -0,0 +1,17 @@
+import * as Interfaces from "@interfaces";
+import * as Utils from "@utils";
+
+const signup: Interfaces.Middlewares.Async = async (req, res, next) => {
+  const body: Interfaces.User = req.body;
+  try {
+    const user = await Utils.createUser(body);
+    if (req.session) {
+      req.session.user = { uid: user.id, username: user.uname };
+    }
+    return res.json({ user });
```

Comment:
use the success function as there's a particular format in which  the response is sent

Reply:
okay.